jujugui/static/gui/src/app/components/search-results/search-results.js

```diff
@@ -228,7 +228,11 @@ YUI.add('search-results', function(Y) {
       }
       this._changeActiveComponent('loading');
       this.setState({ waitingForSearch: true });
-
+      // If there is an exist search request then abort it before starting the
+      // new search.
+      if (this.searchXhr) {
```

Comment:
This appears to get multiple state dispatches (like the other issues we had around this stuff) so it makes multiple searches.